src/components/QuickOpenModal.js

```diff
@@ -136,31 +136,30 @@ export class QuickOpenModal extends Component<Props, State> {
     if (item == null) {
       return;
     }
-    const { selectSource, selectedSource, query, searchType } = this.props;
+    const { selectLocation, selectedSource, query, searchType } = this.props;
     if (this.isSymbolSearch()) {
       if (selectedSource == null) {
         return;
       }
-      selectSource(selectedSource.get("id"), {
-        location: {
-          ...(item.location != null ? { line: item.location.start.line } : {})
-        }
+      selectLocation({
+        sourceId: selectedSource.get("id"),
+        ...(item.location != null ? { line: item.location.start.line } : null)
       });
     } else if (searchType === "gotoSource") {
       const location = parseLineColumn(query);
```

Comment:
does this location include source id info?

Reply:
No, `parseLineColumn` only gets the line and column off of the `query` string argument (i.e. `query` = sourceId:line:column) and returns an object with a separate line and column property. Just thinking, but since `parseLineColumn` just appears in 3 places in the whole code, maybe we should replace it with something more meaningful that builds the whole object (since `query` is technically the full object anyway).